src/main/java/net/earthcomputer/clientcommands/command/PosCommand.java

```diff
@@ -0,0 +1,136 @@
+    // Syntax: "/cpos [to|from <dimension>] [from|to <dimension>] [<pos>]"
+    // "/cpos to x to y", or "/cpos from x from y" are invalid
+    public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher) {
+        dispatcher.register(literal("cpos")
+            .then(argument("pos", blockPos())
+                    .executes(ctx -> convertCoords(ctx.getSource(),
+                            getCBlockPos(ctx, "pos"), null, null)))
+            .then(literal("to").then(argument("targetDimension", dimension())
+                    .then(argument("pos", blockPos())
+                            .executes(ctx -> convertCoords(ctx.getSource(),
+                                    getCBlockPos(ctx, "pos"), null, getCDimensionArgument(ctx, "targetDimension"))))
+                    .then(literal("from").then(argument("sourceDimension", dimension())
+                            .then(argument("pos", blockPos())
+                                    .executes(ctx -> convertCoords(ctx.getSource(),
+                                            getCBlockPos(ctx, "pos"), getCDimensionArgument(ctx, "sourceDimension"), getCDimensionArgument(ctx, "targetDimension"))))
+                            .executes(ctx -> convertCoords(ctx.getSource(),
+                                    ctx.getSource().getPlayer().getBlockPos(), getCDimensionArgument(ctx, "sourceDimension"), getCDimensionArgument(ctx, "targetDimension")))))
+                    .executes(ctx -> convertCoords(ctx.getSource(),
+                            ctx.getSource().getPlayer().getBlockPos(), null, getCDimensionArgument(ctx, "targetDimension")))))
+            .then(literal("from").then(argument("sourceDimension", dimension())
+                    .then(argument("pos", blockPos())
+                            .executes(ctx -> convertCoords(ctx.getSource(),
+                                    getCBlockPos(ctx, "pos"), getCDimensionArgument(ctx, "sourceDimension"), null)))
+                    .then(literal("to").then(argument("targetDimension", dimension())
+                            .then(argument("pos", blockPos())
+                                    .executes(ctx -> convertCoords(ctx.getSource(),
+                                            getCBlockPos(ctx, "pos"), getCDimensionArgument(ctx, "sourceDimension"), getCDimensionArgument(ctx, "targetDimension"))))
+                            .executes(ctx -> convertCoords(ctx.getSource(),
+                                    ctx.getSource().getPlayer().getBlockPos(), getCDimensionArgument(ctx, "sourceDimension"), getCDimensionArgument(ctx, "targetDimension")))))
+                    .executes(ctx -> convertCoords(ctx.getSource(),
+                            ctx.getSource().getPlayer().getBlockPos(), getCDimensionArgument(ctx, "sourceDimension"), null))))
+            .executes(ctx -> convertCoords(ctx.getSource(),
+                    ctx.getSource().getPlayer().getBlockPos(), null, null)));
+    }
```

Comment:
I'd prefer making a custom `ArgumentType` for this command.

Reply:
Okaaaay... trying to understand what you want. Do you want the coupled literal prefix and the dimension type to be a separate `ArgumentType`?

The argument chain is rather complex, I agree, but that is to make sure that a command like `/cpos from the_nether from overworld` can not be entered -- if you go down the `from` path, the second argument can only take a `to`, and vice versa. 

So while I *could* make e.g. a `SelectDimensionArgumentType`, it would lose the ability to prohibit this.

Or do you mean that I should shove the entire thing into a `CPosArgumentType`? I fail to see how that would be an improvement; the `ArgumentType` is intended for "basic" building blocks, and I would not have any easy access to this kind of chaining of methods that describe the syntax. Instead I'd have to resort to more basic parsing approaches.

To be honest, I've already spent much more time on this than I anticipated with my original simple `/cnether` command. At this point I'd rather drop the PR than rewrite this complex chain of arguments as a parser.

Reply:
A custom argument yields more flexibility and control over the command syntax than using the command builder. You could implement an argument type much like `TranslationQueryArgumentType`, but with strict conditions to prevent duplicate clauses.

> To be honest, I've already spent much more time on this than I anticipated with my original simple `/cnether` command. At this point I'd rather drop the PR than rewrite this complex chain of arguments as a parser.

That's understandable. If @Earthcomputer thinks the command is fine as is, I have no problem with it being merged. In any case thanks for your efforts!

Reply:
Yeah I don't see this chain to be a problem. If it were to become more complex in the future for some reason then that would be the case for changing it to an argument type, but I don't see how that's possible in this case.